python/cudf/cudf/core/reshape.py

```diff
@@ -12,6 +12,47 @@
 _axis_map = {0: 0, 1: 1, "index": 0, "columns": 1}
 
 
+def _align_objs(objs, how="outer"):
+    """Align a set of Series or Dataframe objects.
+
+    Parameters
+    ----------
+    objs : list of DataFrame, Series, or Index
+
+    Returns
+    -------
+    A bool for if indexes have matched and a set of
+    reindexed and aligned objects ready for concatenation
+    """
+    # Check if multiindex then check if indexes match. GenericIndex
+    # returns ndarray tuple of bools requiring additional filter.
+    # Then check for duplicate index value.
+    i_objs = iter(objs)
+    first = next(i_objs)
+    if all(isinstance(o.index, cudf.MultiIndex) for o in objs):
+        match_index = all(first.index == rest.index for rest in i_objs)
+    else:
+        match_index = all(all(first.index == rest.index) for rest in i_objs)
```

Comment:
Does `first.index == rest.index` return a scalar boolean? I'd expect that to do an elementwise equals and return an index of boolean dtype. I believe you can do `first.index.equals(rest.index)` instead here.